src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,114 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
 
-export const NewMovie = () => {
+interface Movie {
+  title: string;
+  description: string;
+  imgUrl: string;
+  imdbUrl: string;
+  imdbId: string;
+}
+
+type Props = {
+  onAdd:(tab: Movie) => void,
+};
+
+// eslint-disable-next-line max-len
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+
+  const newMovie = {
+    title,
+    description,
+    imgUrl,
+    imdbUrl,
+    imdbId,
+  };
+
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    onAdd(newMovie);
+    setCount(newCount => newCount + 1);
+    setTitle('');
+    setDescription('');
+    setImgUrl('');
+    setImdbUrl('');
+    setImdbId('');
+  };
+
+  const handleChange = (
+    event: string,
+    setFunc: any,
+  ) => {
+    setFunc(event);
+  };
```

Comment:
```suggestion
  const handleChange = (
    event: ...<...>,
  ) => {
    setMovie({...movie, [event.target.name]: event.target.value });
  };
```
so do not forget to add name prop to each input